drivers/serial/uart_shell.c

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright (c) 2024 Yishai Jaffe <yishai1999@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/uart.h>
+#include <zephyr/shell/shell.h>
+#include <stdlib.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(uart_shell, CONFIG_LOG_DEFAULT_LEVEL);
+
+static int cmd_uart_baudrate(const struct shell *sh, size_t argc, char **argv)
+{
+	char *s_dev_name = argv[1];
+	const struct device *dev;
+	struct uart_config cfg;
+	uint32_t baudrate;
+	int ret;
+
+	dev = device_get_binding(s_dev_name);
+	if (!dev) {
```

Comment:
This does not check if the device passed is an UART, the user is still free to type another device name even with tab-completion.

```suggestion
	if (!dev || !DEVICE_API_IS(uart, dev)) {
```

Reply:
I used the `device_is_uart` function that I already wrote so that there's one place to change if necessary